### PR TITLE
Hotfix/redlining text

### LINKED ISSUE #401: Redlining: edit text does not chnage text color in the map and ancor is not visible
- redlining allows to change existing objects. for point, line, polygon the objects are changed from red to blue, when you click on button pen to edit the object

Bug
- button pen, does not change the color of the text to blue (when you have many textes you do not see which one is choosed)
- the ancor of the text is at the lower-left of the text. But you do not see it.

Solution
- change color of text that you want to edi to blue
- show the ancor point of the text (maybe as a point)

You can test it in demo;

http://version.mapbender3.org/3053/application/mapbender_user

### PR DESCRIPTION
**Pull request by eSlider**
Fix for #401 by @psch. 

@astroidex, @CharlyUnicorn: please make sure this pull-request resolves #401 ticket.

### CODE REVIEW

**Comment by werrolf** [comment]
I'd prefer to merge this into 3.0.5 because it seems like the use case is legacy applications that cannot use the new mapbender-digitizer.

**Comment by astroidex** [comment]
we do not need it in MB 3.0.5. We only use 3.0.6 now and in the future

**Comment by astroidex** [comment]
If redlining is obsolete, then we do not need this hotfix anymore.
 
Quotation from above
"CoreBundle\Element\Redlining is obsolete / superseded by mapbender-digitizer, but LGTM."

**Comment by AxxL** [comment]
Fix for 3.0.6.

**Comment by werrolf** [comment]
Repeated binding / unbinding the keyup event on the input[name='label-text'] seems inefficient, even though the event handler can not be bound several times for repeated _modifyFeature calls (the call to _deactivateControl unbinds the event first).

Event on here: https://github.com/mapbender/mapbender/blob/4642762d7fdbfc73a6e6702f69891117879d155c/src/Mapbender/CoreBundle/Resources/public/mapbender.element.redlining.js#L261
Event off here: https://github.com/mapbender/mapbender/blob/4642762d7fdbfc73a6e6702f69891117879d155c/src/Mapbender/CoreBundle/Resources/public/mapbender.element.redlining.js#L205

The event handling method _writeText already seems to verify (indirectly, by checking for a label in the feature style) that the currently selected feature is of type text. So it may be safe (and clearer) to consolidate event bind / unbind in the activate and deactivate methods, alongside the click event on .redlining-tool.

This should also be a change event, not keyup. Keyup is not always generated on copy&paste interactions and is never generated on form reset.


I also noticed (though not related to changes in this pull) that _newControl will happily leak Openlayers Control objects, which could and should be simply reused per type.

File: src/Mapbender/CoreBundle/Resources/public/mapbender.element.redlining.js [hunk 1]
```diff
@@ -184,33 +191,49 @@
             this.activeControl.activate();
 
         },
-        removeFeature: function(feature){
+        _removeFeature: function(feature){
```
Comment (eSlider) [inline src/Mapbender/CoreBundle/Resources/public/mapbender.element.redlining.js]: Public method change to private in current bug-fix release isn't so cute. :)

File: src/Mapbender/CoreBundle/Resources/public/mapbender.element.redlining.js [hunk 2]
```diff
@@ -184,33 +191,49 @@
...
             this.layer.destroyFeatures([feature]);
         },
+        _removeAllFeatures: function(){
+            $('.geometry-table tr', this.element).remove();
+            this.map.removeLayer(this.layer);
+        },
         _deactivateControl: function(){
+            if(this.selectedFeature) {
+                this.activeControl.unselectFeature(this.selectedFeature);
+                if (this.selectedFeature.style && this.selectedFeature.style.label) {
```
Comment (eSlider) [inline src/Mapbender/CoreBundle/Resources/public/mapbender.element.redlining.js]: What if this isn't this, but that or me and my self?

File: src/Mapbender/CoreBundle/Resources/public/mapbender.element.redlining.js [hunk 3]
```diff
@@ -242,24 +271,41 @@
...
         },
-        _setFeatureStyle: function(){
+        _generateTextStyle: function(label){
             var style = OpenLayers.Util.applyDefaults(null, OpenLayers.Feature.Vector.style['default']);
-            var label = $('input[name=label-text]', this.element).val();
-            style.label = label;
+            if (label) {
+                style.label = label;
+            }
             style.labelAlign = 'lm';
```
Comment (eSlider) [inline src/Mapbender/CoreBundle/Resources/public/mapbender.element.redlining.js]: Put magic numbers into `options`!

Comment (eSlider) [reply src/Mapbender/CoreBundle/Resources/public/mapbender.element.redlining.js]: As default and merge it:  
```js

// Default widget style option
// Surely it's need to be defined on top of the widget.
widget.options.style = {labelXOffset:10, pointRadius: 6};

// And now we merge defaults to style object. Pure magic, not? :)
_.extend(style,widget.options.style);
```

**Pull request merged**